src/flowcean/core/environment/offline.py

```diff
@@ -49,3 +47,53 @@ def stack(self, other: OfflineEnvironment) -> OfflineEnvironment:
         from .stack import StackEnvironment
 
         return StackEnvironment(self, other)
+
+    def to_time_series(
+        self, time_feature: str | dict[str, str]
+    ) -> OfflineEnvironment:
+        """Convert this envrionment to a time series.
+
+        Args:
+            time_feature: The feature in this environment that represents the
+                time vector. Either a string if all series share a common time
+                vector, or a dictionary where the keys are the value features
+                and the values are the corresponding time vector feature names.
+
+        Returns:
+            A OfflineEnvrionment with exactly one sample containg the source
+            environment as a time series.
+        """
+        from flowcean.environments.dataset import Dataset
+
+        # Get the underlaying dataframe
+        data = self.get_data()
+        result_data = pl.DataFrame()
+        # Create the time feature mapping
+        if isinstance(time_feature, str):
+            time_feature = {
+                feature_name: time_feature
+                for feature_name in data.columns
+                if feature_name != time_feature
+            }
+
+        # Convert each feature into a time series
+        for value_feature, t_feature in time_feature.items():
+            result_data = result_data.with_columns(
+                data.select(
+                    [
+                        t_feature,
+                        value_feature,
+                    ]
+                )
+                .rename(
+                    {
+                        t_feature: "time",
+                        value_feature: "value",
+                    }
+                )
+                .to_struct()
+                .implode()
+                .alias(value_feature)
+            )
```

Comment:
You may be able to replace this with a single polars expression...
For example something like this:

```python
data = pl.DataFrame(
    {
        "time_feature": [0, 1, 2, 3],
        "feature_a": [42, 43, 44, 45],
        "feature_b": [3, 2, 1, 0],
    }
)

res = data.select(
    [
        pl.struct(
            pl.col("time_feature").alias("time"),
            pl.col(value_feature).alias("value"),
        )
        .implode()
        .alias(value_feature)
        for value_feature in ["feature_a", "feature_b"]
    ]
)
```

Reply:
That's great. I didn't know about the `pl.struct` method to create structs directly.

Reply:
Can you resolve the conversation so we can merge this one? 😅